internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -113,6 +122,69 @@ func (hr *HypershiftAgentServiceConfigReconciler) getKubeconfigSecret(ctx contex
 	return secret, nil
 }
 
+func (hr *HypershiftAgentServiceConfigReconciler) ensureSpokeAgentInstallCRDs(ctx context.Context, spokeClient client.Client) error {
+	// Fetch agent-install CRDs using in-cluster client
+	localCRDs, err := hr.getAgentInstallCRDs(ctx, hr.Client)
+	if err != nil {
+		return err
+	}
+	if len(localCRDs.Items) == 0 {
+		return pkgerror.New("agent-install CRDs are not available")
+	}
+
+	// Ensure CRDs exist on the spoke cluster
+	for _, item := range localCRDs.Items {
+		crd := item
+		c := crd.DeepCopy()
+		c.ResourceVersion = "" // ResourceVersion should not be set on objects to be created
+		mutate := func() error {
+			c.Spec = crd.Spec
+			c.Annotations = crd.Annotations
+			c.Labels = crd.Labels
+			return nil
+		}
+		result, err1 := controllerutil.CreateOrUpdate(ctx, spokeClient, c, mutate)
+		if err != nil {
+			return pkgerror.Wrapf(err1, "Failed to create CRD '%s' on spoke cluster", crd.Name)
+		}
+		if result != controllerutil.OperationResultNone {
+			hr.Log.Infof("CRD '%s' %s on spoke cluster", crd.Name, result)
+		}
+	}
+
+	// Fetch agent-install CRDs using spoke client
+	spokeCRDs, err := hr.getAgentInstallCRDs(ctx, spokeClient)
+	if err != nil {
+		return err
+	}
+
+	// Remove spoke CRDs that don't exist locally in-cluster
+	for _, item := range spokeCRDs.Items {
+		crd := item
+		existsInCluster := funk.Contains(localCRDs.Items, func(localCRD apiextensionsv1.CustomResourceDefinition) bool {
```

Comment:
Is this comparing the entire CRD object?

Feels like that might be a bit fragile. Is there some simpler comparison we can do that won't fail if something trivial (like a status) happens to be different?

Reply:
/The comparison is actually by CRD name (in the compare func). Any other property worth comparing

Reply:
:facepalm: Missed that entirely.

Nope I think that covers it.